Mathlib/CategoryTheory/Limits/IsLimit.lean

```diff
@@ -0,0 +1,1085 @@
+    P.lift r ≫ (conePointUniqueUpToIso P Q).hom = Q.lift r :=
+  Q.uniq _ _ (by simp)
+#align category_theory.limits.is_limit.lift_comp_cone_point_unique_up_to_iso_hom CategoryTheory.Limits.IsLimit.lift_comp_conePointUniqueUpToIso_hom
+
+@[reassoc (attr := simp)]
+theorem lift_comp_conePointUniqueUpToIso_inv {r s t : Cone F} (P : IsLimit s) (Q : IsLimit t) :
+    Q.lift r ≫ (conePointUniqueUpToIso P Q).inv = P.lift r :=
+  P.uniq _ _ (by simp)
+#align category_theory.limits.is_limit.lift_comp_cone_point_unique_up_to_iso_inv CategoryTheory.Limits.IsLimit.lift_comp_conePointUniqueUpToIso_inv
+
+/-- Transport evidence that a cone is a limit cone across an isomorphism of cones. -/
+def ofIsoLimit {r t : Cone F} (P : IsLimit r) (i : r ≅ t) : IsLimit t :=
+  IsLimit.mkConeMorphism (fun s => P.liftConeMorphism s ≫ i.hom) fun s m => by
+    rw [← i.comp_inv_eq]; apply P.uniq_cone_morphism
+#align category_theory.limits.is_limit.of_iso_limit CategoryTheory.Limits.IsLimit.ofIsoLimit
+
+@[simp]
+theorem ofIsoLimit_lift {r t : Cone F} (P : IsLimit r) (i : r ≅ t) (s) :
+    (P.ofIsoLimit i).lift s = P.lift s ≫ i.hom.Hom :=
+  rfl
+#align category_theory.limits.is_limit.of_iso_limit_lift CategoryTheory.Limits.IsLimit.ofIsoLimit_lift
+
+/-- Isomorphism of cones preserves whether or not they are limiting cones. -/
+def equivIsoLimit {r t : Cone F} (i : r ≅ t) : IsLimit r ≃ IsLimit t
+    where
+  toFun h := h.ofIsoLimit i
+  invFun h := h.ofIsoLimit i.symm
+  left_inv := by aesop_cat
+  right_inv := by aesop_cat
+#align category_theory.limits.is_limit.equiv_iso_limit CategoryTheory.Limits.IsLimit.equivIsoLimit
+
+@[simp]
+theorem equivIsoLimit_apply {r t : Cone F} (i : r ≅ t) (P : IsLimit r) :
+    equivIsoLimit i P = P.ofIsoLimit i :=
+  rfl
+#align category_theory.limits.is_limit.equiv_iso_limit_apply CategoryTheory.Limits.IsLimit.equivIsoLimit_apply
+
+@[simp]
+theorem equivIsoLimit_symm_apply {r t : Cone F} (i : r ≅ t) (P : IsLimit t) :
+    (equivIsoLimit i).symm P = P.ofIsoLimit i.symm :=
+  rfl
+#align category_theory.limits.is_limit.equiv_iso_limit_symm_apply CategoryTheory.Limits.IsLimit.equivIsoLimit_symm_apply
+
+/-- If the canonical morphism from a cone point to a limiting cone point is an iso, then the
+first cone was limiting also.
+-/
+def ofPointIso {r t : Cone F} (P : IsLimit r) [i : IsIso (P.lift t)] : IsLimit t :=
+  ofIsoLimit P
+    (by
+      haveI : IsIso (P.liftConeMorphism t).Hom := i
+      haveI : IsIso (P.liftConeMorphism t) := Cones.cone_iso_of_hom_iso _
+      symm
+      apply asIso (P.liftConeMorphism t))
+#align category_theory.limits.is_limit.of_point_iso CategoryTheory.Limits.IsLimit.ofPointIso
+
+variable {t : Cone F}
+
+theorem hom_lift (h : IsLimit t) {W : C} (m : W ⟶ t.pt) :
+    m =
+      h.lift
+        { pt := W
+          π := { app := fun b => m ≫ t.π.app b } } :=
+  h.uniq
+    { pt := W
+      π := { app := fun b => m ≫ t.π.app b } } m fun b => rfl
```

Comment:
```suggestion
    m = h.lift ⟨W, { app := fun b => m ≫ t.π.app b }⟩ :=
  h.uniq ⟨W, { app := fun b => m ≫ t.π.app b }⟩ m fun b => rfl
```

... and similarly for `hom_desc`.

Not strictly required, but I think it looks nicer.

Reply:
✅